### PR TITLE
Exposes `dataclasses.asdict` to `strawberry.asdict`

### LINKED ISSUE #2170: Expose `dataclasses.asdict` to `strawberry.asdict` and document it
A common use case is to convert types to dicts, which we already support through dataclasses.asdict. 

We should expose `asdict` via the strawberry namespace, so we can make sure there's always a supported way to convert types to dictionaries. And we can also add additional code checks if needed :)

Example here: https://play.strawberry.rocks/?gist=c8e729c52efef8b75f1599c759c069c8

### PR DESCRIPTION
**Pull request by Hazealign**
This Pull Request exposes `dataclasses.asdict(...)` to Strawberry. Also I add some simple tests I expected, and add some document.

## Description

This is a simple changes. Just exposing `dataclasses.asdict(...)` in strawberry. Add some Guide document, and simple test cases too. But it's my first PR of strawberry. So If I did a wrong way, Please review this and tell me how can I improve this. 🙏 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] **Enhancement**/optimization
- [ ] Documentation

Maybe it can be just enhancement. I think many users used `dataclasses.asdict(...)` before we exposed it. So do I.

## Issues Fixed or Closed by This PR

* This Pull Request resolves #2170. 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by botberry** [comment]
Hi, thanks for contributing to Strawberry 🍓!

We noticed that this PR is missing a `RELEASE.md` file. We use that to automatically do releases here on GitHub and, most importantly, to PyPI!

So as soon as this PR is merged, a release will be made 🚀.

Here's an example of `RELEASE.md`:

```markdown
Release type: patch

Description of the changes, ideally with some examples, if adding a new feature.
```

Release type can be one of patch, minor or major. We use [semver](https://semver.org/), so make sure to pick the appropriate type. If in doubt feel free to ask :)


Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to @Hazealign for the PR 👏

Get it here 👉 https://github.com/strawberry-graphql/strawberry/releases/tag/(next)
```

**Comment by patrick91** [comment]
Hi @Hazealign I'll try to review this PR soon, sorry for the wait 😊

**Comment by Hazealign** [comment]
No problem, @patrick91 ! If this will approved, I'll rebase it.

**Comment by Hazealign** [comment]
I'll fix it today. :)

**Review by patrick91 — APPROVED** [review]
Hey @Hazealign, I went and rebased this, hope that's ok!

I've also removed support for a custom factory dict, as I don't think we want to support that for now 😊

**Comment by patrick91** [comment]
Thanks so much for the PR!

**Comment by botberry** [comment]
Thanks for contributing to Strawberry! 🎉 You've been invited to join
the Strawberry GraphQL organisation 😊

You can also request a free sticker by filling this form: https://forms.gle/dmnfQUPoY5gZbVT67

And don't forget to join our discord server: https://strawberry.rocks/discord 🔥

File: docs/guides/convert-to-dictionary.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    name: str
+    age: int
+
+
+# should be {"name": "Lorem", "age": 25}
+user_dict = strawberry.asdict(User(name="Lorem", age=25))
+```
+
+> Note: This function is not recursive, so if you have nested objects, you will
+> need to convert them manually.
```
Comment (jkimbo) [inline docs/guides/convert-to-dictionary.md:24]: My understanding is that `dataclasses.asdict` **is** recursive. Is that not the case? 

This test seems to confirm that as well: https://github.com/strawberry-graphql/strawberry/pull/2417/files#diff-87d39535d5c687a3849638b7599cc65e1baedd3e457fab99f32dc37630e8ee2bR22-R48

Comment (patrick91) [reply docs/guides/convert-to-dictionary.md:24]: you're right! https://github.com/python/cpython/blob/5d868efde9ea08b39357b962a02a0294aa97689c/Lib/dataclasses.py#L1294-L1301

I'll remove this line

**Pull request merged**